### PR TITLE
Event filtering for Materializers

### PR DESCRIPTION
**Pull request by olger**
Initial proposal to enable event filtering in materializers

This consists of a new MaterializerEventFilter interface that is used in the EventMaterializers during replay and live listening. 

This way you can create hour own event filter that matches the deployment situation at hand. (I can imagine that we like another algorithm for kubernetes)

With the filter comes an implementation that is able to filter on the Build information (like version) and the runtime information of the system. 
Runtime and Version information are added to the events in order to use that during filtering. Additionally, the causing command identifier is added to the events. All this information is part of the metadata. 

The runtime information needs some additional work, Build information is now based on string compare but could be extended with actual Version logic. 
Basically this pull request is start for a conversation on the solution and will be merged when the idea and above issues are find. (DONT MERGE)

### CODE REVIEW

**Comment by olger** [comment]
@wketting fully agree, I'll remove the BuildInfo logic on the materializer side. I have the ldea that it *may* help as part of the events for possible support later on. What do you think ?

**Comment by olger** [comment]
@wketting removed it. It's ready for merge if you think it is OK.
